frontend/public/components/storage/shared.ts

```diff
@@ -1,55 +1,120 @@
-import { isCephProvisioner } from '@console/shared/src/utils';
-
 export const cephRBDProvisionerSuffix = 'rbd.csi.ceph.com';
 
 export const snapshotPVCStorageClassAnnotation = 'snapshot.storage.kubernetes.io/pvc-storage-class';
 export const snapshotPVCAccessModeAnnotation = 'snapshot.storage.kubernetes.io/pvc-access-modes';
 export const snapshotPVCVolumeModeAnnotation = 'snapshot.storage.kubernetes.io/pvc-volume-mode';
 
-//See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes for more details
-export const provisionerAccessModeMapping = {
-  'kubernetes.io/no-provisioner': ['ReadWriteOnce'],
-  'kubernetes.io/aws-ebs': ['ReadWriteOnce'],
-  'kubernetes.io/gce-pd': ['ReadWriteOnce', 'ReadOnlyMany'],
-  'kubernetes.io/glusterfs': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
-  'kubernetes.io/cinder': ['ReadWriteOnce'],
-  'kubernetes.io/azure-file': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
-  'kubernetes.io/azure-disk': ['ReadWriteOnce'],
-  'kubernetes.io/quobyte': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
-  'kubernetes.io/rbd': ['ReadWriteOnce', 'ReadOnlyMany'],
-  'kubernetes.io/vsphere-volume': ['ReadWriteOnce', 'ReadWriteMany'],
-  'kubernetes.io/portworx-volume': ['ReadWriteOnce', 'ReadWriteMany'],
-  'kubernetes.io/scaleio': ['ReadWriteOnce', 'ReadOnlyMany'],
-  'kubernetes.io/storageos': ['ReadWriteOnce'],
+export enum VolumeModes {
+  Filesystem = 'Filesystem',
+  Block = 'Block',
+}
+
+export enum AccessModes {
+  RWO = 'ReadWriteOnce',
+  RWX = 'ReadWriteMany',
+  ROX = 'ReadOnlyMany',
+}
+
+type AllowedCombinations = Array<Array<string>>;
+
+const allowedCombinations: AllowedCombinations = [
+  [AccessModes.RWO, VolumeModes.Filesystem],
+  [AccessModes.RWO, VolumeModes.Block],
+  [AccessModes.RWX, VolumeModes.Filesystem],
+  [AccessModes.RWX, VolumeModes.Block],
+  [AccessModes.ROX, VolumeModes.Filesystem],
+  [AccessModes.ROX, VolumeModes.Block],
+  ['All'],
+];
+
+type ProvisionerAccessModeMapping = {
+  [provisioner: string]: AllowedCombinations;
+};
+
+// See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes for more details
+export const provisionerAccessModeMapping: ProvisionerAccessModeMapping = {
+  'kubernetes.io/no-provisioner': [allowedCombinations[0], allowedCombinations[1]],
```

Comment:
Values like `allowedCombinations[0]` are really opaque and have no meaning outside of where they're defined. This really obscures the code to me. I prefer the previous implementation where the values were inline. If we're worried about typos, we should simply create constants.

If we need separate values for Filesystem and Block, create separate maps instead of nested arrays.

Reply:
Here the expected logic is for each provisioner we need to allow some volume mode and access mode combinations, we can't create a map for this volume mode these are the access modes or for these access modes these are the volume modes. 

it should be a combination of both, that why I created a array of supported combinations.

instead of index values 0 or 1 ... can i use combinations directly here like 

[[AccessModes.RWO, VolumeModes.Filesystem], ...]

Reply:
For now, we dont have a complex use case, so let me make it simple:
```
as per UI we select access mode first and then volume mode :

type ProvisionerAccessModeMapping = {
  [provisioner: string]: {
    [accessMode in AccessModes]?: VolumeModes[]
  };
};

export const provisionerAccessModeMapping: ProvisionerAccessModeMapping = {
  'kubernetes.io/no-provisioner': {
    ReadWriteOnce: [VolumeModes.Filesystem, VolumeModes.Block],
  },
}


```

is this make sense?